mesh/accountSettings/views.py

```diff
@@ -87,34 +93,50 @@ def post(self, request):
             )
 
 class SettingsDetailView(View):
+    valid_fields = ['isVerified', 'verificationToken', 'hasContentFilterEnabled', 'displayTheme', 'is2FAEnabled']    
+
     def get(self, request, account_id, *args, **kwargs):
         """
         Handles GET requests when the client fetches for a specific account settings
         """
         try: 
+            fields = request.GET.get('fields')
+            if fields:
+                fields = [field for field in fields.split(',') if field in self.valid_fields]
+            else:
+                fields = self.valid_fields
+            if len(fields) == 0:
+                return JsonResponse({'error': 'Invalid fields'}, status=400)
+            
```

Comment:
Just curious... wasn't the `validate_json_and_required_fields` made for this or is this one a special case that requires us to not use the util function?

Reply:
tdlr: `validate_json_and_required_fields` was not made for this. the two use cases are different.

- `validate_json_and_required_fields` is to validate the request body. It takes in a `json` object, check if the request body has the required fields for the action (i.e. creating account will needs enought information) and returns an indicator saying if the request body is valid for the action.

- The fields checking in this particular case is getting the `fields` param from the request body, then parse it into an array and double-check it with valid fields. This `fields` variable is used to get information from the database. 
    + For example, the account settings have these fields: `twoFactorEnabled`, `theme`, and `notificationSetting`. The `fields` in the request body will indicate how much information the backend is returning, in some cases, we only need partial information. So, instead of returning the whole accountSetting object from the database, we only return the requested fields. If in the request body, `fields="theme,twoFactorEnabled"`, only those information is returned in the response.

Reply:
Okay that makes sense, is there no need to split this into its own function as well or does this seem like it'd be a one-off operation?